### PR TITLE
Upgrade errorprone to 2.19.1

### PR DESCRIPTION
**Pull request by AlexLandau**
## Before this PR
A certain large codebase was blocked from upgrading gradle-baseline because of https://github.com/google/error-prone/issues/3831 (fixed in 2.19.0)

## After this PR

==COMMIT_MSG==
Upgrade Error Prone to 2.19.1 (from 2.18.0)
==COMMIT_MSG==

## Possible downsides?
You may want to review the new/modified checks ([2.19.0](https://github.com/google/error-prone/releases/tag/v2.19.0), [2.19.1](https://github.com/google/error-prone/releases/tag/v2.19.1)) for whether checks should be ignored or promoted.

### CODE REVIEW

**Comment by carterkozak** [comment]
Was the workaround in #2537 not sufficient to unblock?

**Comment by AlexLandau** [comment]
> Was the workaround in #2537 not sufficient to unblock?

It's not the only case where this happens (ImmutableEnumChecker/Immutable is another). I hadn't seen that PR, in any case.

**Comment by carterkozak** [comment]
I’m generally in favor of upgrading, and appreciate this PR, but I’ll need to take a close look through the changes and disable any new checks that are likely to impact a lot of repos since that causes baseline to fall behind.

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java [hunk 1]
```diff
@@ -171,12 +173,23 @@ private static void configureErrorProneOptions(
...
-                            .orElseGet(() -> getNotDisabledErrorproneChecks(
-                                    project, errorProneExtension, javaCompile, maybeSourceSet, errorProneOptions));
-                    return ImmutableList.of(
-                            "-XepPatchChecks:" + Joiner.on(',').join(errorProneChecks.iterator()),
-                            "-XepPatchLocation:IN_PLACE");
+                    Optional<List<String>> specificChecks = getSpecificErrorProneChecks(project);
+                    if (specificChecks.isPresent()) {
+                        List<String> errorProneChecks = specificChecks.get();
+                        return Iterables.concat(
+                                errorProneChecks.stream()
```
Comment (carterkozak) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java:185]: Could you add an inline comment here describing why this addition is necessary?

Comment (AlexLandau) [reply gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java:185]: Done

**Comment by carterkozak** [comment]
Sorry for the delay on this one! I've put together a couple minor modifications, could you take a look over them and merge if you're happy with the changes? #2586
Happy to merge once that goes through

**Comment by AlexLandau** [comment]
> Sorry for the delay on this one! I've put together a couple minor modifications, could you take a look over them and merge if you're happy with the changes? #2586 Happy to merge once that goes through

I merged your changes in (manually, due to the develop-merge issue you mentioned and because this repo only allows squash merges).

**Comment by AlexLandau** [comment]
It looks like the tests are resolving `com.palantir.baseline:baseline-error-prone:latest.release` to `5.11.0`. Meanwhile, running `printVersion` on the PR branch gives `5.7.0-18-gc5fddba.dirty`.

**Comment by AlexLandau** [comment]
I gave up on trying to fix the testkit stuff and just rebased.

**Review by carterkozak — APPROVED** [review]
Thanks!

**Review by bjlaub — APPROVED** [review]

**Pull request merged**